pkg/app/api/api/web_api.go

```diff
@@ -808,6 +845,17 @@ func (a *WebAPI) ApproveStage(ctx context.Context, req *webservice.ApproveStageR
 }
 
 func (a *WebAPI) GetApplicationLiveState(ctx context.Context, req *webservice.GetApplicationLiveStateRequest) (*webservice.GetApplicationLiveStateResponse, error) {
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return nil, err
+	}
+
+	// Check if the requested application belongs to the logged in project.
+	if _, err := a.getApplication(ctx, req.ApplicationId, claims.Role.ProjectId); err != nil {
```

Comment:
It would be nice if we can have some in-memory cache to validate the requested apps, piped...

Reply:
and we can have some common functions like
```
validateApplicationBelongsToProject(appID, projectID) error
validateDeployment...(appID, projectID) error
validatePiped...(appID, projectID) error
```

Reply:
I see, I will give it a try!